app/models/grda_warehouse/auth_policies/policy_provider.rb

```diff
@@ -0,0 +1,127 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'memery'
+
+class GrdaWarehouse::AuthPolicies::PolicyProvider
+  include Memery
+  attr_reader :user
+
+  # TODO: START_ACL remove after ACL migration is complete
+  ###
+  # The legacy permission policy only checks the permissions on a user's roles and assumes the loaded record is visible
+  # to the user already. This is a dangerous assumption so it's required to confirm the record is authorized by
+  # setting legacy_implicitly_assume_authorized_access to true.
+  # Note this is only relevant for users where using_acls is false
+  ###
+  attr_accessor :legacy_implicitly_assume_authorized_access
+  # END_ACL
+
+  def initialize(user)
+    @user = user
+  end
+
+  def for_client(client_or_id)
+    if user.using_acls?
+      client_id = id_from_arg(client_or_id, GrdaWarehouse::Hud::Client)
+      for_client_using_acls(client_id)
+    else
+      # TODO: START_ACL remove after ACL migration is complete
+      legacy_user_role_policy
+      # END_ACL
+    end
+  end
+
+  def for_patient(patient)
+    for_client(patient.client_id)
+  end
+
+  def for_project(project_or_id)
+    if user.using_acls?
+      project_id = id_from_arg(project_or_id, GrdaWarehouse::Hud::Project)
+      for_project_using_acls(project_id)
+    else
+      # TODO: START_ACL remove after ACL migration is complete
+      legacy_user_role_policy
+      # END_ACL
+    end
+  end
+
+  protected
+
+  memoize def for_project_using_acls(project_id)
```

Comment:
@ttoomey does this describe this correctly?  I think my disconnect on the system collection before was that it was just looking for all collections involving the project, regardless of if they were involved in the user's access controls.  That check is performed in `CollectionPolicy`
```suggestion
  # Find all collections that might possibly involve this project,
  # return a collection policy that includes all collections involving the project cross-checked against the user 
  memoize def for_project_using_acls(project_id)
```

Reply:
Yes that's right, the CollectionPolicy has the responsibility for limiting the collections

Maybe I could just reorganize the existing code into `all_collection_ids_for_project` and `all_collection_ids_for_client`? That might make the division of responsibility more clear

```ruby
def for_project_using_acls(project_id)
   # all collections that include the project
   collection_ids = all_collection_ids_for_project(project_id: project_id)
   # policy constrains the collection_ids to only those accessible to the user
   GrdaWarehouse::AuthPolicies::CollectionPolicy.new(user: user, collection_ids: collection_ids)
end

def all_collection_ids_for_project(project_id: project_id)
#...
end

```

Reply:
Yes, that's definitely nicer.  Thanks!